### PR TITLE
Add slides to introduce Project Syn

### PR DESCRIPTION
**Pull request by tobru**
This adds some slides to introduce Project Syn. It should be reusable and updated regularly as Project Syn progresses.

It mainly borrows content from:
* https://syn.tools/syn/getting-started.html
* https://syn.tools/syn/features.html
* https://syn.tools/syn/architecture.html

### CODE REVIEW

File: slides/syntroduction/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.PHONY: all
+all: slides.html slides.pdf
+
+.PHONY: serve
+serve: slides.html
+	caddy
```
Comment (corvus-ch) [inline slides/syntroduction/Makefile]: Using caddy _v2.1.1_, caddy needs the subcommand `run` to run in the foreground or `start` to start as daemon.

File: slides/syntroduction/assets/casts/commodore.cast [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{"version": 2, "width": 190, "height": 26, "timestamp": 1594732914, "env": {"SHELL": "/bin/bash", "TERM": "xterm-256color"}, "title": "Project Syn - Commodore Demo"}
```
Comment (corvus-ch) [inline slides/syntroduction/assets/casts/commodore.cast:1]: Just viewing this on its own, it was too fast for met to grasp what is going on. If this is meant to be accompanied with explanations, this is fine. If it should work on its own, maybe slow it down a bit. Also applies to the other casts.

File: slides/syntroduction/slides_examples.adoc [hunk 3]
```diff
@@ -0,0 +1,357 @@
+// TODO: Add your personal information
```
Comment (corvus-ch) [inline slides/syntroduction/slides_examples.adoc:1]: I guess we do not need the example in here. Also applies to the unused assets.

File: slides/syntroduction/Makefile (hunk 1 above)
Comment (tobru) [reply slides/syntroduction/Makefile]: I haven't put any work into the plumbing, it's plain 1:1 copy of our internal template and I actually focused on the content. If this needs to be fixed, we should do it in our template and update this here when there is a need.

File: slides/syntroduction/slides_examples.adoc (hunk 3 above)
Comment (tobru) [reply slides/syntroduction/slides_examples.adoc:1]: It helps me to write the slides when I can open the example file in the same editor  from the same directory in a second tab and lookup stuff which I forgot.

File: slides/syntroduction/assets/casts/commodore.cast (hunk 2 above)
Comment (tobru) [reply slides/syntroduction/assets/casts/commodore.cast:1]: True that, but it's actually a huge amount of work to record a good looking asciinema show. Looking at it on the terminal using `asciinema play` let's you pause it by hitting `space` and on the web by clicking on the pause button. And that's what I intend to do during presentations. If this turns out to be cumbersome, I might fiddle around with the recording file to make it pause longer on some positions.

**Review by akosma — DISMISSED** [review]
LGTM! However, before merging, I'd remove all unused image files under `assets/images` – this also applies to other subfolders of `assets`, so as to not include unused stuff in this bundle.

File: slides/syntroduction/scripts/lieutenant.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+pe 'curl -H "$LIEUTENANT_AUTH" "http://${LIEUTENANT_URL}/clusters"'
+pe 'kubectl -n lieutenant get cluster'
+pe 'kubectl -n lieutenant get gitrepo'
+
+sleep 3
+
+p '++ Thats it, Lieutenant now has a tenant and a cluster registered. Thanks for watching.'
+
+sleep 1
```
Comment (srueg) [inline slides/syntroduction/scripts/lieutenant.sh]:
```suggestion
sleep 1

```

File: slides/syntroduction/scripts/steward.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+sleep 2
+
+p '++ The install URL is now invalid'
+pe 'kubectl -n lieutenant get cluster ${CLUSTER_ID} -o jsonpath="{.status.bootstrapToken.tokenValid}"'
+
+sleep 5
+# Next page
+clear
+
+p '++ Steward not bootstraps Argo CD which pulls configuration from Cluster Catalog Git Repo'
```
Comment (srueg) [inline slides/syntroduction/scripts/steward.sh]:
```suggestion
p '++ Steward now bootstraps Argo CD which pulls configuration from the Cluster Catalog Git Repo'
```

File: slides/syntroduction/scripts/steward.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+p '++ Lets wait some time to see the Argo CD app to be synced'
+
+sleep 60
+
+pe 'kubectl -n syn get app root -o jsonpath="{.status.sync.status}"'
+
+sleep 5
+
+p '++ Thats it, the cluster is now Project Syn enabled'
```
Comment (srueg) [inline slides/syntroduction/scripts/steward.sh]:
```suggestion
p '++ That's it, the cluster is now Project Syn enabled'

```

File: slides/syntroduction/scripts/steward.sh [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+p '++ Steward not bootstraps Argo CD which pulls configuration from Cluster Catalog Git Repo'
+pe 'kubectl -n syn get pod'
+
+p '++ Lets wait some time to see Pods appear'
+
+sleep 10
+
+pe 'kubectl -n syn get pod'
+
+p '++ Lets wait some time to see the Argo CD app to be synced'
```
Comment (srueg) [inline slides/syntroduction/scripts/steward.sh:57]:
```suggestion
p '++ Let's wait some time to see the Argo CD app to be synced'
```

File: slides/syntroduction/scripts/steward.sh [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+pe 'kubectl -n lieutenant get cluster ${CLUSTER_ID} -o jsonpath="{.status.bootstrapToken.tokenValid}"'
+
+sleep 5
+# Next page
+clear
+
+p '++ Steward not bootstraps Argo CD which pulls configuration from Cluster Catalog Git Repo'
+pe 'kubectl -n syn get pod'
+
+p '++ Lets wait some time to see Pods appear'
```
Comment (srueg) [inline slides/syntroduction/scripts/steward.sh:51]:
```suggestion
p '++ Let's wait some time to see Pods appear'
```

File: slides/syntroduction/scripts/lieutenant.sh [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+. demo-magic.sh -d
+#PROMPT_TIMEOUT=1
+
+# hide the evidence
+clear
+
+p '++ Welcome to Project Syn - Lieutenant Demo'
+sleep 2
+
+p '++ Lets check if the Lieutenant API is healthy and reachable'
```
Comment (srueg) [inline slides/syntroduction/scripts/lieutenant.sh:18]:
```suggestion
p '++ Let's check if the Lieutenant API is healthy and reachable'
```

File: slides/syntroduction/scripts/lieutenant.sh [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+pe 'TENANT_ID=$(curl -s -H "$LIEUTENANT_AUTH" -H "Content-Type: application/json" -X POST --data "{\"displayName\":\"My first Tenant\",\"gitRepo\":{\"url\":\"ssh://git@${GITLAB_ENDPOINT}/${GITLAB_USERNAME}/mytenant.git\"}}" "http://${LIEUTENANT_URL}/tenants" | jq -r ".id")'
+pe 'echo $TENANT_ID'
+pe 'echo https://${GITLAB_ENDPOINT}/${GITLAB_USERNAME}/mytenant'
+
+p '++ This created a Git repository on the above URL for the tenant configuration'
+
+sleep 3
+
+p '++ Lets retrieve the tenants, via API and directly on the cluster'
```
Comment (srueg) [inline slides/syntroduction/scripts/lieutenant.sh:44]:
```suggestion
p '++ Let's retrieve the tenants, via API and directly on the cluster'
```

File: slides/syntroduction/scripts/lieutenant.sh [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+pe 'echo $CLUSTER_ID'
+pe 'echo https://${GITLAB_ENDPOINT}/${GITLAB_USERNAME}/cluster-gitops1'
+
+p '++ Another Git repo got created on the above URL for the cluster catalog'
+
+sleep 5
+## Next Page
+clear
+
+p '++ Lets check the object on the API and on the cluster'
```
Comment (srueg) [inline slides/syntroduction/scripts/lieutenant.sh:73]:
```suggestion
p '++ Let's check the object on the API and on the cluster'
```

File: slides/syntroduction/scripts/lieutenant.sh [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+sleep 2
+
+pe 'curl -H "$LIEUTENANT_AUTH" "http://${LIEUTENANT_URL}/clusters"'
+pe 'kubectl -n lieutenant get cluster'
+pe 'kubectl -n lieutenant get gitrepo'
+
+sleep 3
+
+p '++ Thats it, Lieutenant now has a tenant and a cluster registered. Thanks for watching.'
```
Comment (srueg) [inline slides/syntroduction/scripts/lieutenant.sh:83]:
```suggestion
p '++ That's it, Lieutenant now has a tenant and a cluster registered. Thanks for watching.'
```

File: slides/syntroduction/scripts/steward.sh [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+p '++ Welcome to Project Syn - Steward Demo'
+sleep 2
+
+p '++ This demo assumes a running Lieutenant and a compiled Cluster Catalog'
+
+p '++ Check bootstrap token validity'
+pe 'kubectl -n lieutenant get cluster ${CLUSTER_ID} -o jsonpath="{.status.bootstrapToken.tokenValid}"'
+
+p '++ Lets retrieve the one-time install URL which is only valid once'
```
Comment (srueg) [inline slides/syntroduction/scripts/steward.sh:23]:
```suggestion
p '++ Let's retrieve the one-time install URL which is only valid once'
```

File: slides/syntroduction/scripts/steward.sh [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+pe 'export STEWARD_INSTALL=$(curl -H "$LIEUTENANT_AUTH" -s "http://${LIEUTENANT_URL}/clusters/${CLUSTER_ID}" | jq -r ".installURL")'
+pe 'echo $STEWARD_INSTALL'
+
+sleep 5
+
+p '++ Briefly check the clusters status'
```
Comment (srueg) [inline slides/syntroduction/scripts/steward.sh:30]:
```suggestion
p '++ Briefly check the cluster's status'
```

**Review by srueg — DISMISSED** [review]

**Comment by tobru** [comment]
@srueg Regarding these ' ... I was too  lazy to figure out proper escaping, therefore I left them out. But in the next iteration of these demo scripts I'll add them.

**Review by srueg — APPROVED** [review]

**Pull request merged**